bats_ai/core/utils/guano_utils.py

```diff
@@ -0,0 +1,189 @@
+    date_str = match.group(3)
+    timestamp_str = match.group(4)
+    # extra_data = match.group(5)  # Not used currently
+
+    metadata = {}
+
+    # Extract grid cell ID
+    if cell_id:
+        try:
+            metadata['nabat_grid_cell_grts_id'] = str(int(cell_id))
+        except ValueError:
+            pass
+
+    # Extract date and time
+    if date_str and len(date_str) == 8 and timestamp_str and len(timestamp_str) == 6:
+        try:
+            # Convert YYYYMMDD to date components
+            year = int(date_str[0:4])
+            month = int(date_str[4:6])
+            day = int(date_str[6:8])
+
+            # Convert HHMMSS to time components
+            hour = int(timestamp_str[0:2])
+            minute = int(timestamp_str[2:4])
+            second = int(timestamp_str[4:6])
+
+            # Create datetime object
+            activation_time = datetime(year, month, day, hour, minute, second)
+            metadata['nabat_activation_start_time'] = activation_time
+        except (ValueError, IndexError):
+            pass
+
+    # Extract quadrant if labelName is a valid quadrant
+    if label_name and label_name.upper() in ['SW', 'NE', 'NW', 'SE']:
+        metadata['quadrant'] = label_name.upper()
+
+    return metadata
+
+
+def extract_guano_metadata(file_path: str | Path, check_filename: bool = False) -> dict:
+    """Extract GUANO metadata from a WAV file.
+
+    Args:
+        file_path: Path to the WAV file with GUANO metadata
+
+    Returns:
+        Dictionary containing extracted NABat metadata fields:
+        - nabat_grid_cell_grts_id: str | None
+        - nabat_latitude: float | None
+        - nabat_longitude: float | None
+        - nabat_site_name: str | None
+        - nabat_activation_start_time: datetime | None
+        - nabat_activation_end_time: datetime | None
+        - nabat_software_type: str | None
+        - nabat_species_list: list[str] | None
+        - nabat_comments: str | None
+        - nabat_detector_type: str | None
+        - nabat_unusual_occurrences: str | None
+
+    Raises:
+        Exception: If the file cannot be read or processed
+    """
+    file_path = Path(file_path)
+
+    # Read GUANO metadata from the file
+    gfile = GuanoFile(str(file_path))
+
+    # Extract required NABat fields
+    nabat_fields = {
+        'nabat_grid_cell_grts_id': gfile.get('NABat|Grid Cell GRTS ID', None),
+        'nabat_latitude': gfile.get('NABat|Latitude', None),
+        'nabat_longitude': gfile.get('NABat|Longitude', None),
+        'nabat_site_name': gfile.get('NABat|Site Name', None),
+    }
+
+    # Fix longitude if positive (individuals don't put the - in the longitude)
+    if nabat_fields['nabat_longitude']:
+        try:
+            longitude = float(nabat_fields['nabat_longitude'])
+            if longitude > 0:
+                nabat_fields['nabat_longitude'] = longitude * -1
+            else:
+                nabat_fields['nabat_longitude'] = longitude
```

Comment:
```suggestion
            longitude = abs(float(nabat_fields['nabat_longitude']))
```
I think this is slightly faster to parse, but functionally identical so no action required unless you agree